pulsar-broker/src/main/java/org/apache/pulsar/compaction/TwoPhaseCompactor.java

```diff
@@ -105,7 +105,6 @@ private void phaseOneLoop(RawReader reader,
             return;
         }
         CompletableFuture<RawMessage> future = reader.readNextAsync();
-        scheduleTimeout(future);
```

Comment:
Why did you remove this?

Reply:
the timeout was introduced because we don't have getLastMessageId. but since we introduced getLastMessageId so we know what is the last message id at the time we start compaction, so the reader should be able to read all message till last message id, we don't need to use timeout anymore.

actually timeout makes the compaction behavior *underministic* and hide the problem.

Reply:
actually I think it breaks the case of empty ledger. I will bring it back.

Reply:
brought the timeout back. Although it is not perfect. will look into how to improve this later.